File: sdk/android/library/src/main/java/com/biglabs/solo/signer/library/models/Transaction.java
package com.biglabs.solo.signer.library.models;

import com.google.gson.Gson;

public class Transaction {
    private int id;
    private String method;
    private String[] params;

    public Transaction(String transactionData) {
        this.id = 0;
        this.method = "eth_sendRawTransaction";
        this.params = new String[]{transactionData};
    }

    @Override
    public String toString() {
        return new Gson().toJson(this);
    }
}

File: sdk/android/library/src/main/java/com/biglabs/solo/signer/library/models/Transaction.kt
package com.biglabs.solo.signer.library.models

import com.google.gson.Gson

class Transaction(transactionData: String) {
    private val id: Int = 0
    private val method: String = "eth_sendRawTransaction"
    private val params: Array<String> = arrayOf(transactionData)

    override fun toString(): String {
        return Gson().toJson(this)
    }
}
